modules/sugar/src/main/ts/ephox/sugar/api/dom/Replication.ts

```diff
@@ -22,6 +29,11 @@ const shallowAs = <K extends keyof HTMLElementFullTagNameMap> (original: SugarEl
   const nu = SugarElement.fromTag(tag);
 
   const attributes = Attribute.clone(original);
+  Arr.each(Obj.keys(attributes), (key) => {
+    if (!attributeNameRegex.test(key)) {
+      delete attributes[key];
+    }
+  });
```

Comment:
Makes me wonder if this should go in `Attribute.clone` 🤔

But then looking at _that_ code, perhaps this should inline `clone` rather than do what amounts to 3 iterations over the attribute list (once you look at what `setAll()` does)...

Maybe the whole function could be a single loop over `element.dom.attributes` that only calls `Attribute.set()` if the name is valid. Less pressure on the garbage collector, less looping.